platform/ext/target/nordic_nrf/nrf5340/partition/region_defs.h

```diff
@@ -104,17 +98,25 @@
 
 /* Secure regions */
 #define S_IMAGE_PRIMARY_AREA_OFFSET \
-             (S_IMAGE_PRIMARY_PARTITION_OFFSET + SPU_FLASH_REGION_SIZE)
+                        (S_IMAGE_PRIMARY_PARTITION_OFFSET + BL2_HEADER_SIZE)
 #define S_CODE_START    (S_ROM_ALIAS(S_IMAGE_PRIMARY_AREA_OFFSET))
-#define S_CODE_SIZE     (IMAGE_S_CODE_SIZE - (SPU_FLASH_REGION_SIZE - BL2_HEADER_SIZE))
+#define S_CODE_SIZE     (IMAGE_S_CODE_SIZE)
 #define S_CODE_LIMIT    (S_CODE_START + S_CODE_SIZE - 1)
 
 #define S_DATA_START    (S_RAM_ALIAS(0x0))
 #define S_DATA_SIZE     (TOTAL_RAM_SIZE / 2)
 #define S_DATA_LIMIT    (S_DATA_START + S_DATA_SIZE - 1)
 
-/* CMSE Veneers region */
-#define CMSE_VENEER_REGION_START  (S_IMAGE_PRIMARY_AREA_OFFSET - CMSE_VENEER_REGION_SIZE)
+/* The CMSE veneers shall be placed in an NSC region
+ * which will be placed in a secure SPU region defined
+ * in the beginning of the secure flash region.
+ */
+#define CMSE_VENEER_REGION_SIZE     (0x400)
+#define CMSE_VENEER_REGION_START_ALIGN \
+                . = ALIGN(SPU_FLASH_REGION_SIZE) - (CMSE_VENEER_REGION_SIZE)
```

Comment:
Are these defines only to be used by the linker?
I suppose this file is a standard header to be included in C code, so I'd rather see the #defines not containing the dots :)

Reply:
Instead I'd like to see an absolute value of the address of the start of the veneers region, this is how it is done in other targets, right?

Reply:
> I'd rather see the #defines not containing the dots :)

I can change it to

    #define CMSE_VENEER_REGION_START_ALIGN (ALIGN(SPU_FLASH_REGION_SIZE) - (CMSE_VENEER_REGION_SIZE))

> Instead I'd like to see an absolute value of the address of the start of the veneers region, this is how it is done in other targets, right?

Yes, but we don't want that when we are placing the veneers in code, since we want the linker to place them at the earliest free location that satisfies the requirements. Otherwise we risk wasting even more space.

Reply:
Ah, right. Then, please, document that, because this is different from other targets. Some inline comments explaining the usage of the macros would suffice.

Reply:
Take a look now.